main.rb

```diff
@@ -1,18 +1,7 @@
 require './app'
+require './display_options'
 
 class Main
```

Comment:
- [ ] Based on the requirements of this project, the sole responsibility of main.rb is only to start the code and ask the user for the initial choice of options. We are supposed to thoroughly respect the Single-responsibility Principle of SOLID. Additional methods such as `list_books`, `list_person`, `display_rentals_by_id`, and even `add_rental` should have been implemented in another class.
- Hint: You are not restricted to refactoring your code using only two files. You can create another class for additional methods but each class/method should use the requested YAGNI, KISS, DRY, and SOLID principles.

- [ ] `App class:` No validations should be done in the App class based on the requirements of this project. For example, in the `list_books `method the validation `puts 'No book lists found!' if @books.length.zero?` should have been done in the main.rb file or class/method whose sole responsibility is to do validations.

Reply:
**I have done this based on the reviews**

- [ ] Based on the requirements of this project, the sole responsibility of main.rb is only to start the code and ask the user for the initial choice of options. We are supposed to thoroughly respect the Single-responsibility Principle of SOLID. Additional methods such as `list_books`, `list_person`, `display_rentals_by_id`, and even `add_rental` should have been implemented in another class.
- Hint: You are not restricted to refactoring your code using only two files. You can create another class for additional methods but each class/method should use the requested YAGNI, KISS, DRY, and SOLID principles.

**But in the second requirement,**

- [ ] `App class:` No validations should be done in the App class based on the requirements of this project. For example, in the `list_books `method the validation `puts 'No book lists found!' if @books.length.zero?` should have been done in the main.rb file or class/method whose sole responsibility is to do validations.
I don't see the issues. It works perfectly